src/shared_utilities/Domains.jl

```diff
@@ -184,7 +184,9 @@ end
 A struct holding the necessary information
 to construct a domain, a mesh, a 2d spectral
 element space, and the resulting coordinate field.
-Note that only periodic domains are currently supported.
```

Comment:
I think we need to keep this - because in principle, if we turned on lateral flow, we would be using periodic BC

Alternatively we can state that only independent columns are supported currently (no need for BC in the horizontal)

Reply:
(ClimaCore does not support non-periodic BC, I believe, even for boxes)

Reply:
I added a note saying that only "no lateral flow is supported"